### PR TITLE
Premium Content Block: UI Updates

### PR DESCRIPTION
**Pull request by apeatling**
WIP -- This toolbar or view toggle options need to remain persistent.

Fixes https://github.com/Automattic/view-design/issues/84

**Before:**

94486487-58b55b00-0194-11eb-8614-9dfcf423dce7

**Current Status:**

Screen Shot 2020-10-06 at 3 30 53 PM

### CODE REVIEW

**Comment by apeatling** [comment]
The **Visitor View** | **Subscriber View** toolbar tabs are very important to this block and should remain persistent regardless of which child block is selected. There's no way currently to do this in core Gutenberg.

Two ideas so far:

1. Keep the parent block toolbar visible at the top of the block, and move it up when needed so block toolbars don't overlap
2. Show only specific persistent options on the select parent UI. I'm not sure about this because I don't know if users will understand the difference between this view, and the actual parent toolbar:

Screen Shot 2020-10-06 at 3 46 48 PM (1)

The problem we're trying to solve here is not showing tab or banner UIs on the canvas. Doing so makes creating patterns much harder because the visual is crowded by block UI.

File: apps/editing-toolkit/editing-toolkit-plugin/premium-content/blocks/container/stripe-nudge.js [hunk 1]
```diff
@@ -39,11 +39,11 @@ export const StripeNudge = ( { autosaveAndRedirect, stripeConnectUrl } ) => (
 			{ <Dashicon icon="star-filled" /> }
 			<span className="premium-content-block-nudge__text-container">
 				<span className="premium-content-block-nudge__title">
-					{ __( 'Connect to Stripe to use this block on your site', 'full-site-editing' ) }
+					{ __( 'Connect to Stripe to add premium content to your site.', 'full-site-editing' ) }
 				</span>
 				<span className="premium-content-block-nudge__message">
 					{ __(
-						'This block will be hidden from your visitors until you connect to Stripe.',
+						'Premium content will be hidden from your visitors until you connect to Stripe.',
```
Comment (a8ci18n) [inline apps/editing-toolkit/editing-toolkit-plugin/premium-content/blocks/container/stripe-nudge.js:46]: :ok: This change will be queued for retranslation. We'll use the existing translations in the meantime.

**Comment by jasmussen** [comment]
> The Visitor View | Subscriber View toolbar tabs are very important to this block and should remain persistent regardless of which child block is selected. There's no way currently to do this in core Gutenberg.

This seems like a good use case to improve directly in the core project. 

The closest current equivalent is how reusable blocks work:

reusable

It seems like there's an opportunity to extract and improve how that system works, and perhaps offer it to blocks like these.

**Comment by apeatling** [comment]
Based on the chat, I was thinking about extending the `__experimentalCaptureToolbars` setting for `InnerBlocks` that would provide a breadcrumb for parent blocks. The block toolbar of the parent wouldn't remain persistent, but with a persistent parent block icon it should be visually obvious to the user what happened to it and how to easily get back to it. Something like this:

show-child-toolbar

**Comment by apeatling** [comment]
One issue with this block is that it uses container blocks for the visitor view and the subscriber view. This means the top level container block uses `InnerBlocks` limited to those two blocks. Each of the container blocks then uses `InnerBlocks` to hold all of the content blocks for the respective view.

The issue with this innerBlocks inside innerBlocks issue is you get this awkward middle block container UI when trying to select the parent:



A way to fix this would be to remove the two view blocks and store both of their innerBlocks content in the top level container block. However, there would need to be some way to split the content to only show the blocks for each of the two views. Any ideas on this? Or is it possible to hide this container block UI somehow?

**Comment by jasmussen** [comment]
> One issue with this block is that it uses container blocks for the visitor view and the subscriber view. This means the top level container block uses InnerBlocks limited to those two blocks. Each of the container blocks then uses InnerBlocks to hold all of the content blocks for the respective view.

Is it necessary to surface both those innerblock containers as user editable? It seems like they aren't meant to be interacted with, and could ideally be more or less hidden. It's entirely possible I'm missing an important engineering challenge in suggesting this here, but looking at the Media & Text block, it leverages innerblocks, but surfaces only the necessary controls on the block itself, and on innerblocks in the right container. 

Regarding the mockup in https://github.com/Automattic/wp-calypso/pull/46225#issuecomment-705098313, nice work. 

The toolbar does grow a bit wider with lots of icons asking for attention, which gets a bit more complex with the gray color and the arrow-shaped border. Another approach would be a bit more straight — this is a mockup for image tools, which hasn't been fully implemented yet, but the idea here is that the toolbar for the image is contextual, and when you click the crop button, it enters "edit mode" for the image, where the left pointing chevron then points back to the initial state of the toolbar:

Group 152

I realize that's a little bit different from your use case, where a _child_ block selected inside has its own toolbar competing with that of the parent. 

I think there's still potential in your idea of the _absorbed toolbar with a back button_, but in the name of keeping things moving, perhaps in the near term there's more value in exploring the parent toolbar sticking around, perhaps with a thin gray border around the entire container, even when you have a child block selected — overlap be damned.

**Comment by apeatling** [comment]
We'll be folding these issues into this main PR:

https://github.com/Automattic/view-design/issues/93
https://github.com/Automattic/view-design/issues/94
https://github.com/Automattic/view-design/issues/95
https://github.com/Automattic/view-design/issues/96
https://github.com/Automattic/themes/issues/2281

**Comment by apeatling** [comment]
Rebased.

**Comment by apeatling** [comment]
Since this PR has been broken out, I'm going to close this and open a new PR that only addresses the removal of the double banners. Further improvements will address each separate issue as a new PR.

**Pull request closed without merging**